### ISSUE TITLE
issue about amp and rigid transform

### ISSUE THREAD

**Issue by WeianMao** [jasonkyuyim/se3_diffusion#31]
thanks for this amazing work.
however, in the file 'openfold/utils/rigid_utils.py', I found some matrix multiplication is implemented in a strange way. For example,

    return torch.stack(
        [
            r[..., 0, 0] * x + r[..., 0, 1] * y + r[..., 0, 2] * z,
            r[..., 1, 0] * x + r[..., 1, 1] * y + r[..., 1, 2] * z,
            r[..., 2, 0] * x + r[..., 2, 1] * y + r[..., 2, 2] * z,
        ],
        dim=-1,
    )

why don't using the operator 'A@B'? if u want to disable amp, why not use 'with torch.cuda.amp.autocast(enabled=False):'? thanks.

**Comment by jasonkyuyim** [comment]
Hi, I don't have a good answer. I borrowed the code from openfold which has it implemented like this. PhD life has led me to believe if it ain't broke then don't touch it. That said if there is an optimization or A@B is better then I see no reason why not.
